### ISSUE TITLE
Filter not working with List and also no second time sorting is working

### ISSUE THREAD

**Issue by baba-banshi** [davidegironi/advanceddatagridview#16]
The grid is failed to sort the data if we have sort it using any column it sticks and nothing happens. Filters are not at all working. The list can't sort the data default and so we referred sortable binding list from one of the blog and after that sorting is working but for only once.
Blog for sortable binding list: http://martinwilley.com/net/code/forms/sortablebindinglist.html

**Comment by davidegironi** [comment]
Hello, you have to attache the FilterStringChanged and SortStringChanged events of the grid like exposed on the sample project in order to filter and sort the BindingSource attached to the DataGrid. Hope this helps.

**Comment by baba-banshi** [comment]
Hello, I am using the whole sample project as It is, just using the List instead of DataSet, nothing else changed, these events are also attached and getting fire appropriately.
I have added the file, you can test replacing the FormMain.cs code file using this one.
[FormMain.zip](https://github.com/davidegironi/advanceddatagridview/files/1574473/FormMain.zip)

**Comment by davidegironi** [comment]
Hello, in order to use the Filer and Sort on a binding source you have to use a DataTable.

**Comment by zubyk** [comment]
Hello pals! Thank you for support of my project, you do a great work
I receive many same questions as @baba-banshi  asked (where all that people get my email??)

Sorting/Filter with custom Lists or EntityFramework and all other collections not work because DataSource that you want to use must implement IBindingListView  interface and have Sort and Filter property.

more details you can search here 
https://stackoverflow.com/questions/10074032/datagridview-filter-a-bindingsource-with-a-list-of-object-as-datasource 
here https://social.msdn.microsoft.com/Forums/en-US/1cd860c7-9df5-4dec-bce4-f67b3a823079/bindingsource-sort-doesnt-work?forum=adodotnetentityframework
or ask google in same directions

Dear @davidegironi, can you post this answer in the readme.md or any place where peoples can read it? i receive this question very often

Thanks, Zuby

**Comment by davidegironi** [comment]
Hello @zubyk. I've edit the README.md file. Thank you for this project, you made most of the work, and it great!

**Comment by processor-gis** [comment]
I'm using a datagridview that has over 100k rows and its very slow to get the filtering dialog box to appear.  It usually takes 5 or 6 minutes to open but after that the filtering works fine.  I love this control but is there any possible way to speed it up?  I think the issue might be with populating the unique values in the filtering dialog box.

**Comment by Emporea** [comment]
Is it possible to sort it without knowing the data columns in the first place? If I understand it correctly i need to have BindingSource and declare all datatypes and columns before using a DataTable. 

The Problem is that my DataTable isn't same everytime. I have a simple yet customizable program, where the end user can specifiy a sql connection and the sql query on his own. This makes it impossible to declare the columns in the BindingSource. 

What can I do?

**Comment by davidegironi** [comment]
Hello @Emporea , it should work. Take a look at the dentned reports run form, there custom query are executed and loaded to a filtered datagrid view. https://github.com/davidegironi/dentned/blob/master/DentneD/Forms/FormReportsrun.cs

**Comment by marceloleandro111** [comment]
I want to use filters and sorts in another language? how can i get it? i updated the new version but in execution time we don't have the option... can i set it in code behind or another configuration?

**Comment by davidegironi** [comment]
Hello @marceloleandro111 , take a look at the sample project. You can "inject" the laguage using the SetTranslations method. Like here: https://github.com/davidegironi/advanceddatagridview/blob/0397ee6dcedf97fa68406b31dfa278f5c1f3311a/AdvancedDataGridViewSample/FormMain.cs#L69

**Comment by marceloleandro111** [comment]
Yeaa! that's it. I found this before....thank you!

**Comment by morgp** [comment]
Is it possible to filter data from a button, ie set programmed filtering?  If so how?

**Comment by JackSadowski** [comment]
Possibly related item: [#85](https://github.com/davidegironi/advanceddatagridview/issues/16)

**Comment by export-import20219** [comment]
Hii,
I want to change column header icon programatically. is it possible or not?
Thanks in advance.

**Comment by barawy** [comment]
can you post example with vb.net and access database
mainly how to use the search bar

**Comment by davidegironi** [comment]
Hello @barawy . This library is mainly written in C#, so for the sample project. Unluckily I've no time to dedicate to sample implementation. Buf if you do this, you can poste your sample link here. It would be helpful maybe for other.

**Comment by farshadvl** [comment]
hi. how we can have sum a special column value under that column in this gridview?

**Comment by davidegironi** [comment]
Hello @farshadvl . You can add a custom Row or Colum in your DataTable and use it.

**Comment by farshadvl** [comment]
I mean I want to sum filtered column value .how can I do that?

**Comment by farshadvl** [comment]
> You have to build a custom function to do that. It's not supported natively

I don't have any idea about how to do  that . can u please show us how to code that function?

**Comment by ddveloper1** [comment]
This filter is different from the Excel filter.

If, data = {1,2,3} and I filter only 1
Excel filter shows 1 2 3.
But, this filter shows only 1.
(See attached images)
![excel filter](https://github.com/davidegironi/advanceddatagridview/assets/174328463/a04c61f6-831b-4c08-a5a2-89e2f6ac16dc)
![this filter](https://github.com/davidegironi/advanceddatagridview/assets/174328463/3f0d2655-5d27-4c5b-98d8-5480c7cc2750)

Can you modify it like an Excel filter?

**Comment by davidegironi** [comment]
Your question is not related to this issue, anyway you could use the SetChecklistTextFilterRemoveNodesOnSearchMode method.
Look at the sample project, columns "decimal".
`advancedDataGridView_main.SetChecklistTextFilterRemoveNodesOnSearchMode(advancedDataGridView_main.Columns["decimal"], false);`

**Comment by ddveloper1** [comment]
I confirmed that it works like Excel filter as you said.
Thank you for your response and for creating such a cool control.

**Comment by ClaudioAlbertoSica** [comment]
Hi!

I have a question related not to the advancedDataGridView, but to the advancedDataGridViewSearchToolBar.

I was able to see my ColumnNames whithin the dropdown of the advancedDataGridViewSearchToolBar, BUT it doesn't work. Either I´m missing some configuration, or I did something wrong here.

May be this is a very basic question, but: Is there any other line of code I need to include, to make the Search Bar work, besides the following one?

            advancedDataGridViewSearchToolBar1.SetColumns(advancedDataGridView1.Columns);

Thank you!
